.buildkite/pipeline.yml

```diff
@@ -40,6 +40,12 @@ steps:
     artifact_paths:
       - "**/build/test-results/*/*.xml"
 
+  - label: "Binary compatibility check"
+    if: build.pull_request.base_branch == "trunk"
```

Comment:
Thoughts about this? I think we don't need to run the API ckeck on PRs that are not targeting the `trunk` branch. 

I'm not sure yet what process we would follow when pushing a fix to an older version - whether we would target a tag or a release branch, but we should cover that scenario as well.

Reply:
> Thoughts about this? I think we don't need to run the API ckeck on PRs that are not targeting the trunk branch.

I agree; I don't think we need to cover other branches than `trunk` right now.